packages/azure-http-specs/specs/azure/resource-manager/operation-templates/lro.tsp

```diff
@@ -0,0 +1,251 @@
+    Expected no response body
+    
+    Whether you do polling through AAO, Location or combined, first one will respond with provisioning state "InProgress", second one with "Succeeded".
+    
+    AAO first poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao",
+      "name": "aao",
+      "status" : "InProgress",
+      "startTime": "2024-11-08T01:41:53.5508583+00:00"
+    }
+    ```
+    
+    AAO second poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao",
+      "name": "aao",
+      "status" : "Succeeded",
+      "startTime": "2024-11-08T01:41:53.5508583+00:00",
+      "endTime": "2024-11-08T01:42:41.5354192+00:00"
+    }
+    ```
+    
+    Location first poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/location
+    Expected status code: 202
+    Expected no response body
+    
+    Location second poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/location
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "content": "order1,product1,1"
+    }
+    ```
+    """)
+  @useFinalStateVia("location")
+  export is ArmResourceActionAsync<
+    Order,
+    ExportRequest,
+    ExportResult,
+    LroHeaders = ArmCombinedLroHeaders & Azure.Core.Foundations.RetryAfterHeader
+  >;
+
+  #suppress "deprecated" "legacy"
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header" "legacy"
+  @scenario
+  @scenarioDoc("""
+    Resource DELETE operation.
+    Service returns both Location header on initial request.
+    
+    Expected verb: DELETE
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/orders/order1
+    Expected query parameter: api-version=2023-12-01-preview
+    Expected response status code: 202
+    Expected response header: Location={endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/lro_delete/location
+    Expected no response body
+    
+    Location first poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_delete/location
+    Expected status code: 202
+    Expected no response body
```

Comment:
the polling will always return 202? from recording, it seems always 200. but from guideline, it seems right.

Reply:
Are you sure it's delete + Location? Your [recording](https://github.com/Azure/azure-sdk-assets/blob/7dca1562408eaca118eee8eef0535d9d891b8deb/go/sdk/resourcemanager/compute/armcompute/testdata/recordings/TestVirtualMachinesClient.json#L1942-L1947) seems to poll from AAO, not location.
For AAO, it's always 200. Though for delete + Location, it's 202(poll) and 200/204(success).
See [example flow](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#example-flow).

Reply:
i see.